### PR TITLE
Vue3 Build

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#16981

*Description of changes:*
* Where to begin....
* Upgrade framework to Vue 3, includes router and main.ts major refactors
* Upgrade component library to Vuetify 3 - Includes new vuetify plugin
* Convert projects bundler from Webpack -> Vite - This includes configurations and removal of Webpack/Babel Shtuff
* Updates to Sbc-common-components 4.0.0 version (now 4.0.1) to include minor fixes on issues that were preventing build.
* The jist on the above: Resolves alias conflicts experienced by shared `@` alias in both project and library. Has been resolved in: https://github.com/bcgov/sbc-common-components/pull/337
* Removes obsolete packages (ie tiptap vuetify and stuff no longer required or without vue3 support etc)
* Updated Linting to modern recommendations for Vue, Vuetify and Typescript -- applies fixes (not all): Note this is much stricter than what it was originally but the basic patterns have been set to follow. 

*Vuetify Specific Notes:*
* The goal here is to create somewhat of a reusable assets directory to be shared between projects. So I've removed about 90% of the styling in Base, Layouts and Overrides and have begun to re-add and re-organize with that in mind. So don't be alarmed when you see lots of styling removed, generally appearances are the same. We had a lot of redundant styling, ie styling that could be achieved by helper classes or refactoring the templates to be a more appropriate landmark tag or p tag etc. Once again, trying to keep things generic and centralize styling. 
* Refactored Components to a functional state up to PPR and MHR Searches and History, this includes our App and Dashboard. 

*Additional Notes:*
- With the swap to VITE, we will eventually update our test suites to utilize ViTest, to handle in Task specific to Unit Testing
- I have a checklist of items that i have carried forward to tackle in our Vuetify specific ticket, such as staff payment etc. 
- Lastly, ignore the CI for the time being, we have some updates to make to support the change to VITE and keep in mind this is pointing at a WIP: Feature Branch

- A small look but for those that are interested can certainly build and use Search related features:
Screenshot 2023-10-23 at 2 32 47 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by doug-lovett — APPROVED** [review]
Very impressive for a step. Not too bad to review, lots of the same changes.

File: ppr-ui/src/components/common/PartyForm.vue [hunk 1]
```diff
@@ -15,14 +15,14 @@
             id="person-option"
             class="person-radio"
             label="Individual Person"
-            active-class="selected-radio"
+            false="selected-radio"
```
Comment (dimak1) [inline ppr-ui/src/components/common/PartyForm.vue:25]: Interesting prop `false`

File: ppr-ui/src/components/dialogs/StaffPaymentDialog.vue [hunk 2]
```diff
@@ -5,14 +5,14 @@
...
     <template v-slot:content>
-      <staff-payment-component
-        :staffPaymentData="staffPaymentData"
-        :validate="validating"
-        :displaySideLabel="false"
-        :displayPriorityCheckbox="false"
-        @update:staffPaymentData="onStaffPaymentDataUpdate($event)"
-        @valid="valid = $event"
-      />
+<!--      <StaffPayment-->
```
Comment (dimak1) [inline ppr-ui/src/components/dialogs/StaffPaymentDialog.vue]: Where is the staff payment now?

File: ppr-ui/vite.config.js
Comment (dimak1) [inline ppr-ui/vite.config.js:1]: 👍🏻

**Review by dimak1 — APPROVED** [review]
Very well done! Looks good!

**Comment by dimak1** [comment]
Vue 3 🚀

File: ppr-ui/src/components/common/PartyForm.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/common/PartyForm.vue:25]: Your right, this was done by the linter automation however, I'll look into this!

File: ppr-ui/src/components/dialogs/StaffPaymentDialog.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/dialogs/StaffPaymentDialog.vue]: We have a localized version of it. That being said, i'll probably re-write the shared component in vue3 and port it back. 
As of right now, this is a pending to-do item!

File: ppr-ui/vite.config.js
Comment (cameron-eyds) [reply ppr-ui/vite.config.js:1]: Vite config is pretty good. Very surface level, no so nested and behind the scenes like Webpack was.

File: ppr-ui/src/components/dialogs/StaffPaymentDialog.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/dialogs/StaffPaymentDialog.vue]: Thanks for clarifying!

**Comment by cameron-eyds** [comment]
Just a side note, i started fixing all the linting issues but then reverted them. 
They are helpful when fixing each component individually by understanding what the basic issues are. If i fix them all now, i may miss or forget quick fixes that are done now.

File: ppr-ui/src/components/common/PartyForm.vue (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/common/PartyForm.vue:25]: Thanks for bringing this up. 
As is, this is what the tools are telling me... although i don't quite understand it. 
Will leave as is, for now, when i get to these components will verify it's applying said class as expected or fix if not. 
Very very odd rename of a prop though i agree!

Comment (dimak1) [reply ppr-ui/src/components/common/PartyForm.vue:25]: Yea, ok, as long as it looks as expected. Thanks for looking into it.

**Pull request merged**